cluster-autoscaler/core/static_autoscaler_test.go

```diff
@@ -1280,8 +1281,8 @@ func TestStaticAutoscalerInstanceCreationErrors(t *testing.T) {
 		func(node *apiv1.Node) bool {
 			return false
 		}, nil)
-
-	clusterState = clusterstate.NewClusterStateRegistry(provider, clusterStateConfig, context.LogRecorder, NewBackoff())
+	context.CloudProvider = provider
```

Comment:
This test is confusing as hell, but from what I was able to understand it uses two different CloudProvider mocks, each with different mocks, and then passes first one to the StaticAutoscaler and the second one two the ClusterStateRegistry. Switching the CloudProvider in this context would actually make both StaticAutoscaler and ClusterStateRegistry work on the same copy, which I don't think is intended.

I don't think that unraveling this bundle of joy is in the scope here, but just to be safe I would create a new context and pass it to `NewClusterStateRegistry` instead. This way we keep the original functionality of this test without going too deep into the specifics.

Reply:
Done, thanks!